### ISSUE TITLE
Exploiters fail if OS detection fails

### ISSUE THREAD

**Issue by VakarisZ** [guardicore/monkey#940]
## Describe the bug
If Monkey agent fails OS detection for some reason (ICMP scanning disabled or blocked by firewall) all exploiters also fail. 

## To Reproduce
1. Configure the Monkey to exploit vulnerable machine and disable ping scanner
2. Run the monkey agent
3. Vulnerable machine will not be exploited, since OS is unknown.

## Expected behavior
Partial fix could be: #922 
Exploiters should work if enabled.

## Machine version (please complete the following information):
 - OS: any

**Comment by mssalvatore** [comment]
This looks like a duplicate of https://github.com/guardicore/monkey/issues/921
